### PR TITLE
Libbpf sync 2023-05-24T20:14:39.272Z

### PR DESCRIPTION
**Pull request by danielocfb**
```
Dumping existing libbpf commit signatures...
WORKDIR:          /data/users/muellerd/libbpf
LINUX REPO:       /home/muellerd/local/linux
LIBBPF REPO:      /data/users/muellerd/libbpf
TEMP DIR:         /tmp/tmp.2m8ygyoJou
SUFFIX:           2023-05-24T20-14-39.272Z
BASE COMMIT:      '2ddade3 ("selftests/xsk: Fix munmap for hugepage allocated umem")'
TIP COMMIT:       'fcf1fa2 ("Merge branch 'libbpf: capability for resizing datasec maps'")'
BPF BASE COMMIT:  '71b547f ("bpf: Fix incorrect verifier pruning due to missing register precision taints")'
BPF TIP COMMIT:   'f726e03 ("bpf, sockmap: Test progs verifier error with latest clang")'
SQUASH COMMIT:    75c1f4f1e8dd3eb8cbd3406cb1ddec12dae9e03d
BASELINE TAG:     libbpf-baseline-2023-05-24T20-14-39.272Z
TIP TAG:          libbpf-tip-2023-05-24T20-14-39.272Z
BPF BASELINE TAG: libbpf-bpf-baseline-2023-05-24T20-14-39.272Z
BPF TIP TAG:      libbpf-bpf-tip-2023-05-24T20-14-39.272Z
SQUASH BASE TAG:  libbpf-squash-base-2023-05-24T20-14-39.272Z
SQUASH TIP TAG:   libbpf-squash-tip-2023-05-24T20-14-39.272Z
VIEW TAG:         libbpf-view-2023-05-24T20-14-39.272Z
LIBBPF SYNC TAG:  libbpf-sync-2023-05-24T20-14-39.272Z
PATCHES:          /tmp/tmp.2m8ygyoJou/patches
Updating files: 100% (12390/12390), done.
Switched to a new branch 'libbpf-squash-tip-2023-05-24T20-14-39.272Z'
Picking 'd0fe92 ("tools: bpftool: print netfilter link info")'...
Warning! Cherry-picking 'd0fe92 ("tools: bpftool: print netfilter link info") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Success! All cherry-pick conflicts were resolved for 'd0fe92 ("tools: bpftool: print netfilter link info")'!
Picking '70e7986 ("ELF: fix all "Elf" typos")'...
Picking 'c39028b ("libbpf: btf_dump_type_data_check_overflow needs to consider BTF_MEMBER_BITFIELD_SIZE")'...
Picking '7866fc ("libbpf: Fix comment about arc and riscv arch in bpf_tracing.h")'...
Picking 'bdeeed3 ("libbpf: fix offsetof() and container_of() to work with CO-RE")'...
Picking '10cb86 ("libbpf: Store zero fd to fd_array for loader kfunc relocation")'...
Picking '2b001b ("libbpf: Start v1.3 development cycle")'...
Picking 'cb8edc ("bpf: Support O_PATH FDs in BPF_OBJ_PIN and BPF_OBJ_GET commands")'...
Picking 'f1674d ("libbpf: Add opts-based bpf_obj_pin() API and add support for path_fd")'...
Picking '9d0a233 ("libbpf: Add capability for resizing datasec maps")'...
Commit '29c66ad ("libbpf: Use struct user_pt_regs to define __PT_REGS_CAST() for LoongArch")' is synced into libbpf as:
- e114bd ("libbpf: Use struct user_pt_regs to define __PT_REGS_CAST() for LoongArch")
Skipping '29c66ad ("libbpf: Use struct user_pt_regs to define __PT_REGS_CAST() for LoongArch")' due to unique match...
Commit '06943a ("libbpf: Fix arm syscall regs spec in bpf_tracing.h")' is synced into libbpf as:
- e198fd ("libbpf: Fix arm syscall regs spec in bpf_tracing.h")
Skipping '06943a ("libbpf: Fix arm syscall regs spec in bpf_tracing.h")' due to unique match...
Commit '0a504fa ("libbpf: Document bpf_{btf,link,map,prog}_get_info_by_fd()")' is synced into libbpf as:
- 3fdc11 ("libbpf: Document bpf_{btf,link,map,prog}_get_info_by_fd()")
Skipping '0a504fa ("libbpf: Document bpf_{btf,link,map,prog}_get_info_by_fd()")' due to unique match...
Commit 'c8ee37b ("libbpf: Fix bpf_xdp_query() in old kernels")' is synced into libbpf as:
- 0d7ac2 ("libbpf: Fix bpf_xdp_query() in old kernels")
Skipping 'c8ee37b ("libbpf: Fix bpf_xdp_query() in old kernels")' due to unique match...
Picking 'b5964b ("bpf: Add skb dynptrs")'...
Warning! Cherry-picking 'b5964b ("bpf: Add skb dynptrs") failed, checking if it's non-libbpf files causing problems...
Error! Cherry-picking 'b5964b ("bpf: Add skb dynptrs")' failed, please fix manually and press <return> to proceed...
Picking '05421ae ("bpf: Add xdp dynptrs")'...
Warning! Cherry-picking '05421ae ("bpf: Add xdp dynptrs") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Empty cherry-pick, skipping it...
Picking '66e3a1 ("bpf: Add bpf_dynptr_slice and bpf_dynptr_slice_rdwr")'...
Warning! Cherry-picking '66e3a1 ("bpf: Add bpf_dynptr_slice and bpf_dynptr_slice_rdwr") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Success! All cherry-pick conflicts were resolved for '66e3a1 ("bpf: Add bpf_dynptr_slice and bpf_dynptr_slice_rdwr")'!
Picking '40e1bca ("libbpf: Remove unnecessary ternary operator")'...
Warning! Cherry-picking '40e1bca ("libbpf: Remove unnecessary ternary operator") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Empty cherry-pick, skipping it...
Picking '7832d06 ("libbpf: Remove several dead assignments")'...
Warning! Cherry-picking '7832d06 ("libbpf: Remove several dead assignments") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Empty cherry-pick, skipping it...
Picking '467212 ("libbpf: Cleanup linker_append_elf_relos")'...
Warning! Cherry-picking '467212 ("libbpf: Cleanup linker_append_elf_relos") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Empty cherry-pick, skipping it...
Commit '1eebcb ("libbpf: Implement basic zip archive parsing support")' is synced into libbpf as:
- 9ca6f9 ("libbpf: Implement basic zip archive parsing support")
Skipping '1eebcb ("libbpf: Implement basic zip archive parsing support")' due to unique match...
Commit '434fdcea ("libbpf: Introduce elf_find_func_offset_from_file() function")' is synced into libbpf as:
- 47eb62 ("libbpf: Introduce elf_find_func_offset_from_file() function")
Skipping '434fdcea ("libbpf: Introduce elf_find_func_offset_from_file() function")' due to unique match...
Commit 'c44fd8 ("libbpf: Add support for attaching uprobes to shared objects in APKs")' is synced into libbpf as:
- db2614 ("libbpf: Add support for attaching uprobes to shared objects in APKs")
Skipping 'c44fd8 ("libbpf: Add support for attaching uprobes to shared objects in APKs")' due to unique match...
Picking 'f71f853 ("bpf: Add support for absolute value BPF timers")'...
Warning! Cherry-picking 'f71f853 ("bpf: Add support for absolute value BPF timers") failed, checking if it's non-libbpf files causing problems...
Error! Cherry-picking 'f71f853 ("bpf: Add support for absolute value BPF timers")' failed, please fix manually and press <return> to proceed...
Picking '03b77e ("bpf: Rename __kptr_ref -> __kptr and __kptr -> __kptr_untrusted.")'...
Warning! Cherry-picking '03b77e ("bpf: Rename __kptr_ref -> __kptr and __kptr -> __kptr_untrusted.") failed, checking if it's non-libbpf files causing problems...
Error! Cherry-picking '03b77e ("bpf: Rename __kptr_ref -> __kptr and __kptr -> __kptr_untrusted.")' failed, please fix manually and press <return> to proceed...
Commit 'f8b299b ("libbpf: Add support to set kprobe/uprobe attach mode")' is synced into libbpf as:
- cc7177 ("libbpf: Add support to set kprobe/uprobe attach mode")
Skipping 'f8b299b ("libbpf: Add support to set kprobe/uprobe attach mode")' due to unique match...
Commit '3ecde21 ("libbpf: Fix theoretical u32 underflow in find_cd() function")' is synced into libbpf as:
- 232f42 ("libbpf: Fix theoretical u32 underflow in find_cd() function")
Skipping '3ecde21 ("libbpf: Fix theoretical u32 underflow in find_cd() function")' due to unique match...
Commit '98e678 ("libbpf: Refactor parse_usdt_arg() to re-use code")' is synced into libbpf as:
- c5fe34 ("libbpf: Refactor parse_usdt_arg() to re-use code")
Skipping '98e678 ("libbpf: Refactor parse_usdt_arg() to re-use code")' due to unique match...
Commit '720d93b ("libbpf: USDT arm arg parsing support")' is synced into libbpf as:
- aab5f1 ("libbpf: USDT arm arg parsing support")
Skipping '720d93b ("libbpf: USDT arm arg parsing support")' due to unique match...
Commit '954d1f ("macvlan: Add netlink attribute for broadcast cutoff")' is synced into libbpf as:
- 2c0c92 ("macvlan: Add netlink attribute for broadcast cutoff")
Skipping '954d1f ("macvlan: Add netlink attribute for broadcast cutoff")' due to unique match...
Commit '6018e1f ("bpf: implement numbers iterator")' is synced into libbpf as:
- 32d34a ("bpf: implement numbers iterator")
Skipping '6018e1f ("bpf: implement numbers iterator")' due to unique match...
Commit '5a70f4 ("bpf: Fix a typo for BPF_F_ANY_ALIGNMENT in bpf.h")' is synced into libbpf as:
- c5be1b ("bpf: Fix a typo for BPF_F_ANY_ALIGNMENT in bpf.h")
Skipping '5a70f4 ("bpf: Fix a typo for BPF_F_ANY_ALIGNMENT in bpf.h")' due to unique match...
Commit '27d7fdf ("bpf: use canonical ftrace path")' is synced into libbpf as:
- e8107c ("bpf: use canonical ftrace path")
Skipping '27d7fdf ("bpf: use canonical ftrace path")' due to unique match...
Commit '6cb943 ("libbpf: Ignore warnings about "inefficient alignment"")' is synced into libbpf as:
- c722f7 ("libbpf: Ignore warnings about "inefficient alignment"")
Skipping '6cb943 ("libbpf: Ignore warnings about "inefficient alignment"")' due to unique match...
Commit '5fc13a ("libbpf: Fix relocation of kfunc ksym in ld_imm64 insn.")' is synced into libbpf as:
- fd28ca ("libbpf: Fix relocation of kfunc ksym in ld_imm64 insn.")
Skipping '5fc13a ("libbpf: Fix relocation of kfunc ksym in ld_imm64 insn.")' due to unique match...
Commit '5cbd3fe ("libbpf: Introduce bpf_ksym_exists() macro.")' is synced into libbpf as:
- c401b9 ("libbpf: Introduce bpf_ksym_exists() macro.")
Skipping '5cbd3fe ("libbpf: Introduce bpf_ksym_exists() macro.")' due to unique match...
Commit '08ff1c9 ("bpf, docs: Libbpf overview documentation")' is synced into libbpf as:
- 2db620 ("bpf, docs: Libbpf overview documentation")
Skipping '08ff1c9 ("bpf, docs: Libbpf overview documentation")' due to unique match...
Commit 'a506d6c ("libbpf: Fix ld_imm64 copy logic for ksym in light skeleton.")' is synced into libbpf as:
- 5b45c9 ("libbpf: Fix ld_imm64 copy logic for ksym in light skeleton.")
Skipping 'a506d6c ("libbpf: Fix ld_imm64 copy logic for ksym in light skeleton.")' due to unique match...
Commit '01dc26c ("libbpf: Explicitly call write to append content to file")' is synced into libbpf as:
- 753e4d ("libbpf: Explicitly call write to append content to file")
Skipping '01dc26c ("libbpf: Explicitly call write to append content to file")' due to unique match...
Commit 'a18f72 ("libbpf: Rename RELO_EXTERN_VAR/FUNC.")' is synced into libbpf as:
- a5464a ("libbpf: Rename RELO_EXTERN_VAR/FUNC.")
Skipping 'a18f72 ("libbpf: Rename RELO_EXTERN_VAR/FUNC.")' due to unique match...
Commit '708cdc ("libbpf: Support kfunc detection in light skeleton.")' is synced into libbpf as:
- 68cd7c ("libbpf: Support kfunc detection in light skeleton.")
Skipping '708cdc ("libbpf: Support kfunc detection in light skeleton.")' due to unique match...
Commit '68b0486 ("bpf: Create links for BPF struct_ops maps.")' is synced into libbpf as:
- 077bf7 ("bpf: Create links for BPF struct_ops maps.")
Skipping '68b0486 ("bpf: Create links for BPF struct_ops maps.")' due to unique match...
Commit '8d1608 ("libbpf: Create a bpf_link in bpf_map__attach_struct_ops().")' is synced into libbpf as:
- c6893d ("libbpf: Create a bpf_link in bpf_map__attach_struct_ops().")
Skipping '8d1608 ("libbpf: Create a bpf_link in bpf_map__attach_struct_ops().")' due to unique match...
Commit 'aef56f ("bpf: Update the struct_ops of a bpf_link.")' is synced into libbpf as:
- 055cbd ("bpf: Update the struct_ops of a bpf_link.")
Skipping 'aef56f ("bpf: Update the struct_ops of a bpf_link.")' due to unique match...
Commit '912dd4 ("libbpf: Update a bpf_link with another struct_ops.")' is synced into libbpf as:
- e51cda ("libbpf: Update a bpf_link with another struct_ops.")
Skipping '912dd4 ("libbpf: Update a bpf_link with another struct_ops.")' due to unique match...
Commit '809a69 ("libbpf: Use .struct_ops.link section to indicate a struct_ops with a link.")' is synced into libbpf as:
- 465a73 ("libbpf: Use .struct_ops.link section to indicate a struct_ops with a link.")
Skipping '809a69 ("libbpf: Use .struct_ops.link section to indicate a struct_ops with a link.")' due to unique match...
Picking 'f1cb92 ("libbpf: Ensure print callback usage is thread-safe")'...
Warning! Cherry-picking 'f1cb92 ("libbpf: Ensure print callback usage is thread-safe") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Empty cherry-pick, skipping it...
Picking 'd08ab8 ("libbpf: Fix double-free when linker processes empty sections")'...
Warning! Cherry-picking 'd08ab8 ("libbpf: Fix double-free when linker processes empty sections") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Empty cherry-pick, skipping it...
Picking 'd6e6286 ("libbpf: disassociate section handler on explicit bpf_program__set_type() call")'...
Warning! Cherry-picking 'd6e6286 ("libbpf: disassociate section handler on explicit bpf_program__set_type() call") failed, checking if it's non-libbpf files causing problems...
Looks like only non-libbpf files have conflicts, ignoring...
Empty cherry-pick, skipping it...
Commit 'e0aee1 ("libbpf: Don't enforce unnecessary verifier log restrictions on libbpf side")' is synced into libbpf as:
- a22abb ("libbpf: Don't enforce unnecessary verifier log restrictions on libbpf side")
Skipping 'e0aee1 ("libbpf: Don't enforce unnecessary verifier log restrictions on libbpf side")' due to unique match...
Commit '47a71c ("bpf: Add log_true_size output field to return necessary log buffer size")' is synced into libbpf as:
- 88004d ("bpf: Add log_true_size output field to return necessary log buffer size")
Skipping '47a71c ("bpf: Add log_true_size output field to return necessary log buffer size")' due to unique match...
Commit '94e55c ("libbpf: Wire through log_true_size returned from kernel for BPF_PROG_LOAD")' is synced into libbpf as:
- c2fe7a ("libbpf: Wire through log_true_size returned from kernel for BPF_PROG_LOAD")
Skipping '94e55c ("libbpf: Wire through log_true_size returned from kernel for BPF_PROG_LOAD")' due to unique match...
Commit '097d80 ("libbpf: Wire through log_true_size for bpf_btf_load() API")' is synced into libbpf as:
- 2f0156 ("libbpf: Wire through log_true_size for bpf_btf_load() API")
Skipping '097d80 ("libbpf: Wire through log_true_size for bpf_btf_load() API")' due to unique match...
Commit 'd54730 ("bpf: Introduce opaque bpf_refcount struct and add btf_record plumbing")' is synced into libbpf as:
- 44f59e ("bpf: Introduce opaque bpf_refcount struct and add btf_record plumbing")
Skipping 'd54730 ("bpf: Introduce opaque bpf_refcount struct and add btf_record plumbing")' due to unique match...
Commit '3055dd ("libbpf: misc internal libbpf clean ups around log fixup")' is synced into libbpf as:
- 0db753 ("libbpf: misc internal libbpf clean ups around log fixup")
Skipping '3055dd ("libbpf: misc internal libbpf clean ups around log fixup")' due to unique match...
Commit 'f70916 ("libbpf: report vmlinux vs module name when dealing with ksyms")' is synced into libbpf as:
- 6fd310 ("libbpf: report vmlinux vs module name when dealing with ksyms")
Skipping 'f70916 ("libbpf: report vmlinux vs module name when dealing with ksyms")' due to unique match...
Commit '05b6f76 ("libbpf: improve handling of unresolved kfuncs")' is synced into libbpf as:
- e6cc30 ("libbpf: improve handling of unresolved kfuncs")
Skipping '05b6f76 ("libbpf: improve handling of unresolved kfuncs")' due to unique match...
Commit 'c5e6474 ("libbpf: move bpf_for(), bpf_for_each(), and bpf_repeat() into bpf_helpers.h")' is synced into libbpf as:
- 2293c2 ("libbpf: move bpf_for(), bpf_for_each(), and bpf_repeat() into bpf_helpers.h")
Skipping 'c5e6474 ("libbpf: move bpf_for(), bpf_for_each(), and bpf_repeat() into bpf_helpers.h")' due to unique match...
Commit '94dccb ("libbpf: mark bpf_iter_num_{new,next,destroy} as __weak")' is synced into libbpf as:
- 294c85 ("libbpf: mark bpf_iter_num_{new,next,destroy} as __weak")
Skipping '94dccb ("libbpf: mark bpf_iter_num_{new,next,destroy} as __weak")' due to unique match...
Commit 'd0fe92 ("tools: bpftool: print netfilter link info")' is synced into libbpf as:
- LINUX_27c037 ("tools: bpftool: print netfilter link info")
Skipping 'd0fe92 ("tools: bpftool: print netfilter link info")' due to unique match...
Commit '70e7986 ("ELF: fix all "Elf" typos")' is synced into libbpf as:
- LINUX_c1a674 ("ELF: fix all "Elf" typos")
Skipping '70e7986 ("ELF: fix all "Elf" typos")' due to unique match...
Rewrite 7b7bda2d76020e275daa44bd4668f5a1ed954a6c (12/12) (47 seconds passed, remaining 0 predicted)
Ref 'refs/heads/libbpf-squash-tip-2023-05-24T20-14-39.272Z' was rewritten
Ref 'refs/heads/libbpf-squash-base-2023-05-24T20-14-39.272Z' was rewritten
Rewrite 32543182244f4e73b1c46c0af3b6362d2ce9c46e (8/11) (1 seconds passed, remaining 0 predicted)
Ref 'refs/heads/libbpf-squash-tip-2023-05-24T20-14-39.272Z' was rewritten
Ref 'refs/heads/libbpf-squash-base-2023-05-24T20-14-39.272Z' was rewritten
/tmp/tmp.2m8ygyoJou/patches/0000-cover-letter.patch
/tmp/tmp.2m8ygyoJou/patches/0001-tools-bpftool-print-netfilter-link-info.patch
/tmp/tmp.2m8ygyoJou/patches/0002-ELF-fix-all-Elf-typos.patch
/tmp/tmp.2m8ygyoJou/patches/0003-libbpf-btf_dump_type_data_check_overflow-needs-to-co.patch
/tmp/tmp.2m8ygyoJou/patches/0004-libbpf-Fix-comment-about-arc-and-riscv-arch-in-bpf_t.patch
/tmp/tmp.2m8ygyoJou/patches/0005-libbpf-fix-offsetof-and-container_of-to-work-with-CO.patch
/tmp/tmp.2m8ygyoJou/patches/0006-libbpf-Store-zero-fd-to-fd_array-for-loader-kfunc-re.patch
/tmp/tmp.2m8ygyoJou/patches/0007-libbpf-Start-v1.3-development-cycle.patch
/tmp/tmp.2m8ygyoJou/patches/0008-bpf-Support-O_PATH-FDs-in-BPF_OBJ_PIN-and-BPF_OBJ_GE.patch
/tmp/tmp.2m8ygyoJou/patches/0009-libbpf-Add-opts-based-bpf_obj_pin-API-and-add-suppor.patch
/tmp/tmp.2m8ygyoJou/patches/0010-libbpf-Add-capability-for-resizing-datasec-maps.patch
Switched to a new branch 'libbpf-sync-2023-05-24T20-14-39.272Z'
Applying: tools: bpftool: print netfilter link info
Applying: ELF: fix all "Elf" typos
Applying: libbpf: btf_dump_type_data_check_overflow needs to consider BTF_MEMBER_BITFIELD_SIZE
Applying: libbpf: Fix comment about arc and riscv arch in bpf_tracing.h
Applying: libbpf: fix offsetof() and container_of() to work with CO-RE
Applying: libbpf: Store zero fd to fd_array for loader kfunc relocation
Applying: libbpf: Start v1.3 development cycle
Applying: bpf: Support O_PATH FDs in BPF_OBJ_PIN and BPF_OBJ_GET commands
Applying: libbpf: Add opts-based bpf_obj_pin() API and add support for path_fd
Applying: libbpf: Add capability for resizing datasec maps
Updating files: 100% (80382/80382), done.
Switched to branch 'libbpf-tip-2023-05-24T20-14-39.272Z'
[libbpf-sync-2023-05-24T20-14-39.272Z 17ce8e] sync: latest libbpf changes from kernel
 2 files changed, 2 insertions(+), 2 deletions(-)
SUCCESS! 10 commits synced.
Verifying Linux's and Github's libbpf state
Switched to a new branch 'libbpf-view-2023-05-24T20-14-39.272Z'
Rewrite fcf1fa29c8ea75bf104c35ce29b65ce2ba6a6a9d (3/3) (20 seconds passed, remaining 0 predicted)
Ref 'refs/heads/libbpf-view-2023-05-24T20-14-39.272Z' was rewritten
Rewrite 123c4cfbeb24ba09df2c92cc0dce27482517a048 (1/1) (0 seconds passed, remaining 0 predicted)
Ref 'refs/heads/libbpf-view-2023-05-24T20-14-39.272Z' was rewritten
Comparing list of files...
Comparing file contents...
Great! Content is identical!
Cleaning up...
Updating files: 100% (80382/80382), done.
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
Deleted branch libbpf-baseline-2023-05-24T20-14-39.272Z (was 2ddade3).
Deleted branch libbpf-tip-2023-05-24T20-14-39.272Z (was fcf1fa2).
Deleted branch libbpf-bpf-baseline-2023-05-24T20-14-39.272Z (was 71b547f).
Deleted branch libbpf-bpf-tip-2023-05-24T20-14-39.272Z (was f726e03).
Deleted branch libbpf-squash-base-2023-05-24T20-14-39.272Z (was 1be07e6).
Deleted branch libbpf-squash-tip-2023-05-24T20-14-39.272Z (was a5f89b).
Deleted branch libbpf-view-2023-05-24T20-14-39.272Z (was d57ed3).
DONE.
```

### CODE REVIEW

File: src/libbpf.c [hunk 1]
```diff
@@ -9409,10 +9437,103 @@
...
+	if (element_sz <= 0 || (size - var->offset) % element_sz != 0) {
+		pr_warn("map '%s': cannot be resized, element size (%d) doesn't align with new total size (%u)\n",
+			bpf_map__name(map), element_sz, size);
+		return -EINVAL;
+	}
+
+	/* create a new array based on the existing array, but with new length */
+	nr_elements = (size - var->offset) / element_sz;
+	new_array_id = btf__add_array(btf, array->index_type, array->type, nr_elements);
+	if (new_array_id < 0)
```
[the remark this answers is not in the document]
Comment (danielocfb) [reply src/libbpf.c:9493]: Should be fixed with https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/commit/?id=4c857a719bf9c5ddbcf3bd92398632041bff50d0. Next sync.

[the remark this answers is not in the document]
Comment (danielocfb) [reply src/libbpf.c:9493]: Actually I cherry picked it now.

**Comment by anakryiko** [comment]
We'll get https://patchwork.kernel.org/project/netdevbpf/patch/20230525232248.640465-1-deso@posteo.net/ on the next sync. Otherwise looks good, thanks!

**Comment by anakryiko** [comment]
oh, you added that selftest fix as temporary patch, nice

**Pull request merged**